packages/smarthr-ui/src/components/Button/ButtonWrapper.tsx

```diff
@@ -53,49 +63,91 @@ export function ButtonWrapper({
   className,
   ...rest
 }: Props) {
-  const wrapperClassName = useMemo(() => {
+  const innerRef = useRef<HTMLElement>(null)
+  // HINT: squareは
+  //  null: Buttonのレンダリング前
+  //  boolean: レンダリング後
+  const [square, setSquare] = useState<null | boolean>(null)
+
+  useEffect(() => {
+    if (innerRef.current) {
+      // HINT: prefix, suffixが存在せず、かつicon,svg,imgのいずれかが単一でbodyに含まれるButtonの場合true
+      setSquare(!!innerRef.current.querySelector(ICON_BUTTON_SELECTOR))
+    }
+  }, [children])
+
+  const classNames = useMemo(() => {
     const generate = wrapperClassNameGenerator({
```

Comment:
nits: 展開した方が generate が何か気にしなくて良くなるかも?（汎用っぽい名前だから何かと衝突するのかしら

```suggestion
    const { button, anchor, loader } = wrapperClassNameGenerator({
```

Reply:
あー、ほかだと基本的に全部キーにして値を返してたから、そのアタリ違うことを表したかった...とか？
展開しちゃって良い気がしますね